pyartcd/pyartcd/pipelines/ocp4.py

```diff
@@ -0,0 +1,96 @@
+import os
+
+import click
+from aioredlock import LockError
+
+from pyartcd import exectools, locks
+from pyartcd.cli import cli, pass_runtime, click_coroutine
+from pyartcd.runtime import Runtime
+
+
+async def sync_repo_to_s3_mirror(local_dir: str, s3_path: str, dry_run: bool):
+    if s3_path.startswith('/pub/openshift-v4/clients') or \
+            s3_path.startswith('/pub/openshift-v4/amd64') or \
+            s3_path.startswith('/pub/openshift-v4/arm64') or \
+            s3_path.startswith('/pub/openshift-v4/dependencies'):
+        raise Exception(
+            f'Invalid location on s3 ({s3_path}); these are virtual/read-only locations on the s3 '
+            'backed mirror. Qualify your path with /pub/openshift-v4/<brew_arch_name>/ instead.')
+
+    # Sync is not transactional. If we update repomd.xml before files it references are populated,
+    # users of the repo will get a 404. So we run in three passes:
+    # 1. On the first pass, exclude files like repomd.xml and do not delete any old files.
+    #    This ensures that we  are only adding new rpms, filelist archives, etc.
+    # 2. On the second pass, include only the repomd.xml.
+    base_cmd = ['aws', 's3', 'sync', '--no-progress', '--exact-timestamps']
+    if dry_run:
+        base_cmd.append('--dryrun')
+
+    cmd = base_cmd + [
+        '--exclude', '*/repomd.xml', local_dir,
+        f's3://art-srv-enterprise{s3_path}'  # Note that s3_path has / prefix.
```

Comment:
I know the original groovy code doesn't do that, but could we add a check to ensure `s3_path` always starts with a `/`?

Reply:
Done in cd4100ecee5272706a5b33cebd1ec880e3f65c1b